packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,216 @@
+   * @param props {LITAuthenticateProps} Authentication params, only `context` is required
+   * @returns {Promise<LitSessionSigsMap>} Authenticated session material
+   * @throws {Not Authenticated} if authentication operations fail this error is thrown
+   */
+  authenticate = async (
+    props: LitAuthenticateProps<C>
+  ): Promise<LitUserMetadata> => {
+    if (!this.session) {
+      // runs authentication logic
+      await this._doAuthentication(props);
+    }
+
+    // check on internal state for authentication status
+    if (!this.session) {
+      throw new Error("Not Authenticated");
+    }
+
+    return this.session;
+  };
+
+  getAuthDetails = async (): Promise<LitUserMetadata> => {
+    this._checkInternals();
+    return this._authContext as LitSessionSigsMap;
+  };
+
+  getAddress = async () => {
+    this._checkInternals();
+    return this.signer?.getAddress() as Promise<Address>;
+  };
+
+  signMessage = async (msg: Uint8Array | string) => {
+    this._checkInternals();
+
+    return this.signer?.signMessage(msg) as Promise<Hex>;
+  };
+
+  signTypedData = (params: SignTypedDataParams) => {
+    this._checkInternals();
+
+    return this.signer?._signTypedData(
+      params.domain as TypedDataDomain,
+      params.types as any,
+      params.message
+    ) as Promise<Hex>;
+  };
+
+  private _checkInternals() {
+    if (!this._authContext) {
+      throw new Error("Not Authenticated");
+    }
+
+    if (!this.signer) {
+      throw new Error("Signer is not initalized, did you call authenticate?");
+    }
+  }
+
+  /**
+   * Runs the Lit Protocol authentication operations for a given piece of authentication material
+   *
+   * AuthMethod -> authenticates the auth material and signs a session.
+   *
+   * SessionSigsMap -> uses the session to create a signer instance.
+   *
+   * For more information on Lit Authentication see below:
+   *
+   * https://developer.litprotocol.com/v3/sdk/authentication/overview
+   * @param props {LitAuthenticationProps<C>} properties for configuring authentication operations
+   */
+  private async _doAuthentication(props: LitAuthenticateProps<C>) {
+    /**
+     * Check if the object is structured as an auth method
+     * if so, we sign the session key with the auth method
+     * as the auth material. Otherwise, if a session signature
+     * is provided, then we skip this step.
+     */
+    if (Object.keys(props.context).indexOf("accessToken") > 0) {
+      const resourceAbilities = [
+        {
+          resource: new LitPKPResource("*"),
+          ability: LitAbility.PKPSigning,
+        },
+      ];
+      const sessionKeypair = props.sessionKeypair || generateSessionKeyPair();
+      const chain = props.chain || "ethereum";
+      const chainInfo = ALL_LIT_CHAINS[chain];
+      // @ts-expect-error - chainId is not defined on the type
+      const chainId = chainInfo.chainId ?? 1;
```

Comment:
is there a reason this gives a lint error if youre expecting this field?

Reply:
we have a type issue here. I could `any` cast it but i used a `ts-error` to explain why its omitted.

Reply:
Update:  I added an explicit cast as the type is a union to only a single type of the union as we are only dealing with EVM chains. I was able to remove the ignore statement. Added a link to some documentation to the authentication props.